empmgmt-servlet/src/main/java/com/neha/empmgmt/servlet/EmployeeServlet.java

```diff
@@ -1,38 +1,143 @@
+			doDelete(request, response);
+		} else if (request.getMethod().equalsIgnoreCase(HttpMethod.PUT) || 
+				(request.getParameter("action") != null && request.getParameter("action").equalsIgnoreCase("update"))){
+			doDelete(request, response);
+		} else if (request.getMethod().equalsIgnoreCase(HttpMethod.PUT)) {
+			doGet(request, response);
+		} else {
+			if(request.getParameter("action") != null){
+				throw new RuntimeException(request.getMethod() + " is not supported for action = "+request.getParameter("action"));
+			}
+			throw new RuntimeException(request.getMethod() + " is not supported");
+		}
+	}
 
 	@Override
-	protected void doDelete(HttpServletRequest arg0, HttpServletResponse arg1) throws ServletException, IOException {
-		// TODO Auto-generated method stub
-		System.out.println("Testin githuub commit");
-		super.doDelete(arg0, arg1);
+	protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		// Write logic to handle delete of an employee / employees
+		// Check to see if id is present in the url = > deleteById
+		// If id is not present => deleteAll
+		Boolean isDeleted = false;
+		if (request.getParameter("id") != null) {
+			isDeleted = employeeService.deleteById(Integer.parseInt(request.getParameter("id")));
+
+		} else {
+			//TODO: instead of deleteAll, display the message that employee not found/could not delete
+			isDeleted = employeeService.deleteAll();
+		}
+		// TODO: Display message in UI that the EmployeeWithID/Employees deleted successfully
+		// OR show the updated result by doing a findAll().
 	}
 
 	@Override
-	protected void doGet(HttpServletRequest arg0, HttpServletResponse arg1) throws ServletException, IOException {
-		// TODO Auto-generated method stub
-		super.doGet(arg0, arg1);
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		// Write logic to handle get employee / employees
+		// Check to see if id is present in the url => getById
+		// if id is not present => findAll
+		// return appropriate response;
+
+		if (request.getParameter("id") != null) {
+			Employee employee = employeeService.findById(Integer.parseInt(request.getParameter("id")));
+			// TODO :return appropriate response;
+		} else {
+			List<Employee> employees = employeeService.findAll();
+			// TODO :return appropriate response;
+		}
 	}
 
 	@Override
-	protected void doPost(HttpServletRequest arg0, HttpServletResponse arg1) throws ServletException, IOException {
-		// TODO Auto-generated method stub
-		super.doPost(arg0, arg1);
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		// write logic to post employee / employees
+		// get all the employee parameters and make employee object
+		// call save()
+		Boolean isSaved = false;
+		Employee employee = new Employee();

```

Comment:
Employee creation code in POST and PUT is being repeated. You can move this to a separate method in the same class, lets name it - Public Employee populateEmployee(Request req) which will simply return the populated employee from the request.